.travis.yml

```diff
@@ -26,12 +26,22 @@ env:
         - DMD=2.071.2.s1 DIST=xenial F=production
         - DMD=2.071.2.s1 DIST=xenial F=devel
 
+# Don't build tags already converted to D2
+if: NOT tag =~ ^.*\+d2^
+
 install: beaver dlang install
 
 script: BEAVER_DOCKER_VARS="F AFTER_SCRIPT" beaver dlang make
 
-after_success:
-    - test "$AFTER_SCRIPT" = "1" && beaver run ci/codecov.sh
-    - test "$AFTER_SCRIPT" = "1" &&
-        BEAVER_DOCKER_VARS="OCEAN_D2_USER OCEAN_D2_PASS"
-        beaver run ci/d2tags.sh
+after_success: test "$AFTER_SCRIPT" = "1" && beaver run ci/codecov.sh
+
+jobs:
+    include:
+        - stage: D2 Release
+          if: tag IS present
+          # We override it, otherwise is inherited from the first matrix row
+          env: DIST=xenial
+          # before_install and install are inherited
+          script: beaver dlang d2-release
+          # Overridden to NOP, otherwise is inherited
+          after_success: true
```

Comment:
Maybe we should just move codecov script to `script` phase to make things simpler here?

Reply:
I think even if we move codecov to script, this is an accident waiting to happen, so I would be inclined to keep the comments. To be honest travis stages seem like an ugly hack for now.

And having codecov as a `after_success` script is slightly more correct because if it fails it won't make the build fail.